src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -204,6 +225,110 @@ send_exception_message(const char* msg)
 
 }
 
+//////////////////////////////////////////////////////////////
+// xrt_graph C++ API implementations (xrt_graph.h)
+//////////////////////////////////////////////////////////////
+namespace xrt {
+
+graph::
+graph(const xrt::device& device, const xrt::uuid& xclbin_id, const std::string& name)
+  : handle(open_graph(device, xclbin_id, name))
+{}
+
+void
```

Comment:
Perhaps for such simple things you could have 4 lines in 1 line.
Also I am wondering whether all these simple functions could no be placed directly in a `.hpp` instead to allow inlining. Or do you want to have the `handle` hidden from the public headers?

Reply:
I am left leaning and so it shall remain :-)
I believe open_graph is internal.